airflow/operators/python.py

```diff
@@ -176,7 +178,7 @@ def execute(self, context: Context) -> Any:
 
         return return_value
 
-    def determine_kwargs(self, context: Mapping[str, Any]) -> Mapping[str, Any]:
+    def determine_kwargs(self, context: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
```

Comment:
This does not need to change either.

Reply:
I think I know why MyPy was complaining. The problem was that if only .py file changed, Mypy did not really use .pyi file to determine the actual type - we have a number of cases where type in .pyi overrides the types and if it does not change and MyPy does not see the .pyi file it will not know the type is overridden in .pyi file when you run it incrementally. 

I will add a small change to the mypy pre-commit to see if there are .pyi files corresponding to changed .py files and add them to the list of files if they are missing.

Reply:
No. Actually the problem is that previously mypy has not detected the modifications we've done to op_kwargs.

```
        if self.templates_dict:
            op_kwargs['templates_dict'] = self.templates_dict
```

is not possible for Mapping (Mapping is immutable). Seems it "accidentally" worked because our Mapping was really a Dict. But the right solution is to create a new mapping out of it.

Reply:
Yep. This works:
````
        op_kwargs: Dict[str, Any] = dict()
        op_kwargs.update(self.op_kwargs)
        if self.templates_dict:
            op_kwargs['templates_dict'] = self.templates_dict

```

Reply:
Actualy it's both - @uranusjr. I just have a case where `--all-files` produces different result than just changes from this commit, so likely the hypothesis of having to add .pyi is right. Will experiment a bit.